app/config/FeatureSwitches.scala

```diff
@@ -16,24 +16,45 @@
 
 package config
 
+import com.google.inject.ImplementedBy
 import org.apache.commons.lang3.BooleanUtils
 import play.api.Configuration
 import play.api.mvc.Request
 
-case class FeatureSwitches(featureSwitchConfig: Configuration) {
+import javax.inject.{Inject, Singleton}
+
+@ImplementedBy(classOf[FeatureSwitchesImpl])
+trait FeatureSwitches {
+
+  def isCl402Enabled: Boolean
+  def isTemporalValidationEnabled(implicit request: Request[_]): Boolean
+}
+
+@Singleton
+class FeatureSwitchesImpl(featureSwitchConfig: Configuration) extends FeatureSwitches {
+
+  @Inject
+  def this(appConfig: AppConfig) = this(appConfig.featureSwitches)
+
+  val isCl402Enabled: Boolean = isEnabled("cl402")
```

Comment:
If the switch is going to be used to exclude the endpoint from the OAS in poroduction (and add something about it being Test Only in external test), then:

1.  it doesn't look like the OAS re-writer code is in this API. You might want to pull it in from BSAS
2. the OasFeatureRewriter uses the feature name as a string so you might not need this method